testing/test_session.py

```diff
@@ -253,3 +256,34 @@ def pytest_sessionfinish():
     """)
     res = testdir.runpytest("--collect-only")
     assert res.ret == EXIT_NOTESTSCOLLECTED
+
+
+@pytest.mark.parametrize("path", ["root", "{relative}/root", "{environment}/root"])
+def test_rootdir_option_arg(testdir, path):
+    if 'relative' in path:
+        path = path.format(relative=os.getcwd())
+    if 'environment' in path:
+        os.environ['PY_ROOTDIR_PATH'] = os.getcwd()
+        path = path.format(environment='$PY_ROOTDIR_PATH')
+
+    rootdir = testdir.mkdir("root")
+    rootdir.mkdir("tests")
+    testdir.makepyfile("""
+        import os
+        def test_one():
+            assert 1
+    """)
+
+    result = testdir.runpytest("--rootdir={}".format(os.path.expandvars(path)))
```

Comment:
As I commented earlier we need to remove this `os.path.expandvars` call from here.